WordPress/Classes/Networking/PlansRemote.swift

```diff
@@ -54,9 +56,173 @@ private func mapPlansResponse(response: AnyObject) throws -> (activePlan: Plan,
+        guard let (planFeatures, _) = cachedPlanFeaturesWithDate() else { return nil}
+
+        return planFeatures
+    }
+    
+    /// @return An optional tuple containing a collection of cached plan features and the date when they were fetched
+    private func cachedPlanFeaturesWithDate() -> (PlanFeatures, NSDate)? {
+        if let cacheFileURL = cacheFileURL,
+            let path = cacheFileURL.path {
+                
+            if NSFileManager.defaultManager().fileExistsAtPath(path) {
+                do {
+                    let attributes = try NSFileManager.defaultManager().attributesOfItemAtPath(path)
+                    if let modificationDate = attributes[NSFileModificationDate] as? NSDate {
+                        if cacheDateIsValid(modificationDate) {
+                            if let response = NSData(contentsOfURL: cacheFileURL) {
+                                let json = try NSJSONSerialization.JSONObjectWithData(response, options: [])
+                                return (try mapPlanFeaturesResponse(json), modificationDate)
+                            }
+                        }
+                    }
+                }
+                catch {}
+            }
+        }
+        
+        return nil
+    }
+    
+    private func cacheDateIsValid(date: NSDate) -> Bool {
+        return NSCalendar.currentCalendar().isDateInToday(date)
+    }
+    
+    private func fetchPlanFeatures(success: PlanFeatures -> Void, failure: ErrorType -> Void) {
+        let endpoint = "plans/features"
+        let path = pathForEndpoint(endpoint, withVersion: ServiceRemoteRESTApiVersion_1_2)
+        
+        api.GET(path,
+            parameters: nil,
+            success: {
+                [weak self] requestOperation, response in
+                self?.cacheResponseData(requestOperation.responseData)
+                do {
+                    try success(mapPlanFeaturesResponse(response))
+                } catch {
+                    failure(error)
+                }
+            }, failure: {
+                _, error in
+                failure(error)
+        })
+    }
+    
+    private var cacheFileURL: NSURL? {
+        guard let cacheDirectory = NSFileManager.defaultManager().URLsForDirectory(.CachesDirectory, inDomains: .UserDomainMask).first else { return nil }
+        
+        return cacheDirectory.URLByAppendingPathComponent(cacheFilename)
+    }
+    
+    private func cacheResponseData(responseData: NSData?) {
+        guard let responseData = responseData else { return }
+        guard let cacheFileURL = cacheFileURL else { return }
+        
+        responseData.writeToURL(cacheFileURL, atomically: true)
+    }
+}
+
+private func mapPlanFeaturesResponse(response: AnyObject) throws -> PlanFeatures {
+    guard let json = response as? [[String: AnyObject]] else {
+        throw PlansRemote.Error.DecodeError
+    }
+    
+    var features = [PlanID: [PlanFeature]]()
```

Comment:
Making a note here regarding the discussion we just had – this is due to features being duplicated where they have different descriptions for different plans (e.g. storage amounts).